cognite/neat/core/data_classes/transformation_rules.py

```diff
@@ -37,39 +41,59 @@ class URL(BaseModel):
 
 class Resource(BaseModel):
     # Solution model
-    class_name: str = Field(alias="Class")
-    description: str = Field(alias="Description", default="")
+    description: str = Field(alias="Description", max_length=1024, default=None)
 
-    # Solution CDF resource
-    cdf_resource_type: str = Field(alias="Resource Type")
+    # Solution CDF resource, it is not needed when working with FDM, this is only for
+    # Classic CDF data model
+    cdf_resource_type: str = Field(alias="Resource Type", default=None)
 
-    # Solution model advance configuration
+    # Advance data modeling: Keeping track if Resource got deprecated or not
     deprecated: bool = Field(default=False)
     deprecation_date: Optional[datetime] = Field(alias="deprecationDate", default=None)
     replaced_by: str = Field(alias="replacedBy", default=None)
-    similar_to: HttpUrl = Field(alias="similarTo", default=None)
-    similarity_score: float = Field(alias="similarityScore", default=None)
-    equal_to: HttpUrl = Field(alias="equalTo", default=None)
+
+    # Advance data modeling: Relation to existing resources for purpose of mapping
+    source: HttpUrl = Field(
+        alias="Source", description="Source of information for given entity, e.g. CIM", default=None
+    )
+    source_entity_name: str = Field(
+        alias="Source Entity Name", description="Closest entity in source, e.g. Substation", default=None
+    )
+    match_type: str = Field(
+        alias="Match Type", description="Type of match between source entity and one being defined", default=None
+    )
+    comment: str = Field(alias="Comment", description="Comment about mapping", default=None)
+    issues: List[str] = Field(default=None, description="Storing list of pydantic validation issues")
+    valid: bool = Field(default=True, description="Indicates whether resource is valid or not")
 
     @validator(
         "deprecated",
         "deprecation_date",
         "replaced_by",
-        "similar_to",
-        "similarity_score",
-        "equal_to",
-        "description",
+        "source",
+        "source_entity_name",
+        "match_type",
+        "comment",
         pre=True,
     )
     def replace_float_nan_with_default(cls, value, field):
         if isinstance(value, float) and math.isnan(value):
             return field.default
         return value
 
+    class Config:
+        allow_population_by_field_name = True
+        anystr_strip_whitespace = True
+
+
+class_id_compliance_regex = r"^([a-zA-Z]+[a-zA-Z0-9]+[._-]{0,1}[a-zA-Z0-9]+)+$"
+
 
 class Class(Resource):
+    class_id: str = Field(alias="Class", min_length=1, max_length=255)
+    class_name: str = Field(alias="Name", min_length=1, max_length=255, default=None)
```

Comment:
You can add this to the top of the file 
```python
from pydantic import constr

ExternalId = constr(min_length=1, max_length=255)
```
Then

```suggestion
    class_id: ExternalId = Field(alias="Class", )
    class_name: ExternalId = Field(alias="Name", default=None)
```